### ISSUE TITLE
feat: add the `sign` feature

### ISSUE THREAD

**Issue by iqdecay** [FuelLabs/forc-wallet#7]
```bash
# Sign a hash         
forc wallet sign <wallet number> <hash>
```

**Comment by adlerjohn** [comment]
There should also be functionality to sign a message, not just a message hash.

**Comment by adlerjohn** [comment]
Additionally, signing with this command should not sign the raw hash, but MUST first domain-separate the hash. Standardizing the domain separator can be bikeshed.

**Comment by kayagokalp** [comment]
While #19 is approaching the finish line, I did some experiments and look the [`sign_transaction`](https://github.com/FuelLabs/fuels-rs/blob/7fb18dfeeb12519fc7e6bd87e8da003d2e8dc9ee/packages/fuels-signers/src/wallet.rs#L454) at SDK, related to the signing to plan this. If I understood correctly the signing process will look like the following (at least for starters). 

1. Generate deployment transaction from `forc-deploy` and copy the transaction id generated.
2. Execute something like `forc-wallet sign <the transaction id>` which generates the signature and we copy the signature back to `forc-deploy`.
3. `forc-deploy` will generate `Witness` from the signature we took from `forc-wallet` and add the witness (like [this](https://github.com/FuelLabs/fuels-rs/blob/7fb18dfeeb12519fc7e6bd87e8da003d2e8dc9ee/packages/fuels-signers/src/wallet.rs#L469-L475) part) to the transaction before submitting it.

Does this sound correct (at least conceptually)?
